in_toto/models/layout.py

```diff
@@ -120,15 +99,21 @@ def read_from_file(filename='root.layout'):
   @staticmethod
   def read(data):
     """Static method to instantiate a new Layout from a Python dictionary """
+    signatures = data.get('signatures', [])
     steps = []
-    for step_data in data.get("steps"):
+
+    if 'signed' not in data:
+        data['signed'] = _Layout()
```

Comment:
I think you could leave this out (which would also prevent coverage decline), as you do the same thing in `common.Metablock`'s constructor anyway, right? (https://github.com/in-toto/in-toto/blob/c1e0cf122f2e1faf6583784517a23cb752a0fc77/in_toto/models/common.py#L56-L57)

Reply:
Sure will do, but I'm less worried about losing 0.01% of coverage than having an increasingly convoluted constructor.

 I do think that this is bringing up the bad coupling patterns that we have on our constructors though. This spaghetti code is just reflecting the fact that we'd rather use .read and inline the json rather than instantiating metadata objects all throughout the code.